### PR TITLE
move dev deployment to script

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Move dev deployment to script.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/hack/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+if [[ ! -f ${2} ]]; then
+    echo "File not found!"
+    exit 1
+fi
+
+HELM_EXTRA_ARGS=${3:-""}
+VALUES_FILE=$(realpath ${2})
+cd "$(dirname "$0")/../../"
+
+helm upgrade --install --wait --timeout 300 ${HELM_EXTRA_ARGS} --values ${VALUES_FILE} --namespace nginx-ingress-controller nginx-ingress-controller ./config/nginx-ingress-controller/
```
Comment (alvaroaleman) [inline api/hack/deploy.sh:20]: Is the namespace change here intended?

File: .drone.yml [hunk 2]
```diff
@@ -92,69 +92,18 @@ pipeline:
...
-      --set=kubermatic.rbac.image.tag=${DRONE_COMMIT}
-    image: kubeciio/helm
-    pull: true
+    commands:
+    - export VAULT_ADDR=https://vault.loodse.com/
+    - export VAULT_TOKEN=$(vault write --format=json auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_SECRET_ID | jq .auth.client_token -r)
+    - vault kv get -field=kubeconfig dev/seed-clusters/dev.kubermatic.io > kubeconfig
+    - vault kv get -field=values.yaml dev/seed-clusters/dev.kubermatic.io > values.yaml
+    - export KUBECONFIG=$(pwd)/kubeconfig
+    - export EXTRA_ARGS="--tiller-namespace=kubermatic-installer --set=kubermatic.isMaster=false --set=kubermatic.controller.image.tag=${DRONE_COMMIT_SHA} --set=kubermatic.api.image.tag=${DRONE_COMMIT_SHA} --set=kubermatic.rbac.image.tag=${DRONE_COMMIT_SHA}"
```
Comment (alvaroaleman) [inline .drone.yml]: why `kubermatic.isMaster=false`?

File: containers/helm/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM alpine
+
+RUN apk add --no-cache ca-certificates curl
+RUN curl -Lo /usr/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.12.2/bin/linux/amd64/kubectl && \
```
Comment (alvaroaleman) [inline containers/helm/Dockerfile:4]: I'll mark this day in my calender

File: containers/helm/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    chmod +x /usr/bin/kubectl
+
+RUN curl -Lo /tmp/helm.tar.gz https://storage.googleapis.com/kubernetes-helm/helm-v2.11.0-linux-amd64.tar.gz && \
```
Comment (alvaroaleman) [inline containers/helm/Dockerfile]: Just do `curl -L $URL|tar -xvz`

File: api/hack/deploy.sh (hunk 1 above)
Comment (mrIncompetent) [reply api/hack/deploy.sh:20]: Yes, it now follows the namespace we use in our [docs](https://docs.kubermatic.io/installation/install_kubermatic/_manual/)

I'll take care of the necessary DNS changes

File: containers/helm/Dockerfile (hunk 4 above)
Comment (mrIncompetent) [reply containers/helm/Dockerfile]: Done with https://github.com/kubermatic/kubermatic/pull/2400/commits/cac711ebc4e3e75277a482afe805421c8c04144a

File: .drone.yml (hunk 2 above)
Comment (mrIncompetent) [reply .drone.yml]: Fixed with https://github.com/kubermatic/kubermatic/pull/2400/commits/cac711ebc4e3e75277a482afe805421c8c04144a

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**